modules/effects/arrow-effects-data/src/main/kotlin/arrow/effects/typeclasses/MonadDefer.kt

```diff
@@ -44,8 +44,7 @@ interface MonadDefer<F> : MonadThrow<F>, Bracket<F, Throwable> {
 
   fun lazy(): Kind<F, Unit> = delay { }
 
-  fun <A> delayUnsafe(f: () -> Either<Throwable, A>): Kind<F, A> =
-    defer { f().fold({ raiseError<A>(it) }, { just(it) }) }
+  fun <A> delayUnsafe(f: () -> Either<Throwable, A>): Kind<F, A> = deferUnsafe(f)
```

Comment:
The other way around, the old implementation should call the new implementation :D My bad for not being clear about it!

Reply:
omg completely my fault :see_no_evil: